src/node/time.rs

```diff
@@ -0,0 +1,58 @@
+use std::sync::{Arc, RwLock};
+
+/// Manages timestamps across the system.
+///
+/// Clones always agree on the underlying timestamp and updating one affects all other instances.
+#[derive(Clone, Debug)]
+pub struct TimestampManager {
+    /// The latest timestamp that has already been used.
+    last_timestamp: Arc<RwLock<u64>>,
+}
```

Comment:
Given the above, maybe it makes sense to indicate the used unit? Not sure whether we want to change the method names, but maybe mention in the main doc comment and method doc comments that we're using seconds here.